codec/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java

```diff
@@ -134,6 +134,8 @@ public ByteBuf cumulate(ByteBufAllocator alloc, ByteBuf cumulation, ByteBuf in)
     private boolean singleDecode;
     private boolean decodeWasNull;
     private boolean first;
+    private boolean callingChildDecode;
+    private boolean handlerRemovedPending;
```

Comment:
2 booleans was a bit simpler but I will just use a `byte` if we want to be more definitive about memory usage.